### PR TITLE
feat(ui): add source/target field from the mapping details pane

### LINKED ISSUE #1989: UI_REACT: add "add new mapped field" (either source or target) widget in the mapping details view

### PR DESCRIPTION
**Pull request by riccardo-forina**
Fixes #1989 (ENTESB-13782)

Also never show notifications about a mapping as toasts, and make notifications in the mapping details pane dismissible, with a more compact look and with better wording (ENTESB-13737).

### CODE REVIEW

File: ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  fields,
+  ariaLabelTypeAhead,
+  placeholderText,
+  ...props
+}) => {
+  const { state, toggle, toggleOff } = useToggle(false);
+
+  const renderOptions = (fields: IAddFieldTypeaheadField[]) => {
+    return fields.map((f, index) => (
+      <SelectOption
```
Comment (pleacu) [inline ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx:34]: The select options for fields should list the document name (unselectable) first followed by selectable fields in that document as opposed listing the document for each field redundantly.

File: ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        variant={"typeahead"}
+        ariaLabelTypeAhead={ariaLabelTypeAhead}
+        onToggle={toggle}
+        isExpanded={state}
+        placeholderText={placeholderText}
+        onFilter={filterFields}
+        onSelect={onSelect}
+        maxHeight={300}
+      >
+        {renderOptions(fields)}
```
Comment (pleacu) [inline ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx:83]: Need to add properties and constants as selectable fields

File: ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx (hunk 1 above)
Comment (riccardo-forina) [reply ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx:34]: It does
image

File: ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx (hunk 2 above)
Comment (riccardo-forina) [reply ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx:83]: Ops, right, thanks!

File: ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx (hunk 1 above)
Comment (pleacu) [reply ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx:34]: Only one JSONInstanceSource - like this:
sc1

Comment (riccardo-forina) [reply ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx:34]: That's not how grouped selects should be done with PF4. Check the example here https://www.patternfly.org/v4/documentation/react/components/select#grouped-single

Comment (pleacu) [reply ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx:34]: It's close.  Anyway - this can be tackled as a PF4 grouped-single as another issue.

Comment (riccardo-forina) [reply ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx]: There now, PF4 style:
image

File: ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx (hunk 2 above)
Comment (riccardo-forina) [reply ui/packages/atlasmap/src/UI/AddFieldTypeahead.tsx:98]: https://github.com/atlasmap/atlasmap/pull/2071/files#diff-ed5af85d7b40907087a3ae8f3ea5bd57R118-R124

**Pull request merged**